### PR TITLE
Use retrofit to do network requests

### PR DESCRIPTION
**Pull request by fourlastor**
#22 

Basically I passed retrofit inside the various managers and got rid of whatever was doing the requests before. EZ

### CODE REVIEW

File: app/src/main/java/com/connfa/model/managers/EventManager.java [hunk 1]
```diff
@@ -21,18 +23,18 @@ public EventManager(Context context, DrupalClient client) {
...
-        return null;
+    public boolean storeResponse(Event.Holder requestResponse, String tag) {
+        return false;
     }
 
     @Override
-    protected boolean storeResponse(Event.Holder requestResponse, String tag) {
-        return false;
+    protected Observable<Event.Holder> doFetch(ConnfaRepository repository) {
+        throw new UnsupportedOperationException("You can't do this.");
```
Comment (fourlastor) [inline app/src/main/java/com/connfa/model/managers/EventManager.java:37]: ¯\_(ツ)_/¯

File: app/src/main/java/com/connfa/model/UpdatesManager.java [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Action;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Function;
 import timber.log.Timber;
 
-public class UpdatesManager {
+public class UpdatesManager implements Closeable {
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/UpdatesManager.java:25]: 😱 woooo

File: app/src/main/java/com/connfa/model/UpdatesManager.java [hunk 3]
```diff
@@ -58,176 +57,157 @@ public static int convertEventIdToEventModePos(int eventModePos) {
...
+    public UpdatesManager(Context context) {
         this.context = context;
-        this.client = client;
-        this.preferencesManager = PreferencesManager.create(context);
         updateListeners = new ObserverHolder<>();
     }
 
     public void startLoading(@NotNull final UpdateCallback callback) {
-        new AsyncTask<Void, Void, List<Integer>>() {
+        final ConnfaRepository repository = ((ConnfaApplication) context.getApplicationContext()).getConnfaRepository();
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/UpdatesManager.java]: take out the finals plz

File: app/src/main/java/com/connfa/model/UpdatesManager.java [hunk 4]
```diff
@@ -58,176 +57,157 @@ public static int convertEventIdToEventModePos(int eventModePos) {
...
 
-    public void registerUpdateListener(DataUpdatedListener listener) {
-        this.updateListeners.registerObserver(listener);
+    private Function<SynchronousItemManager, ObservableSource<Integer>> fetchManagerData(final Integer id, final ConnfaRepository repository, final ILAPIDBFacade facade) {
+        return new Function<SynchronousItemManager, ObservableSource<Integer>>() {
+            @Override
+            public ObservableSource<Integer> apply(SynchronousItemManager manager) throws Exception {
+                //noinspection unchecked
+                return manager.fetch(repository, facade)
+                        .map(new Function() {
```
Comment (rock3r) [inline app/src/main/java/com/connfa/model/UpdatesManager.java:110]: Extract method

File: app/src/main/java/com/connfa/service/ConnfaRepository.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        ConnfaService service = retrofit.create(ConnfaService.class);
+        return new ConnfaRepository(service);
+    }
+
+    private ConnfaRepository(ConnfaService service) {
+        this.service = service;
+    }
+
+    public Observable<Updates> updates() {
+        return service.updates().map(new Function<ApiUpdates, Updates>() {
```
Comment (rock3r) [inline app/src/main/java/com/connfa/service/ConnfaRepository.java]: Extract method plz

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java [hunk 6]
```diff
@@ -35,49 +36,47 @@
...
-import junit.framework.Assert;
-
-import android.support.annotation.NonNull;
-
 import java.lang.reflect.Field;
 import java.lang.reflect.Modifier;
 import java.util.HashMap;
 import java.util.Map;
 
+import junit.framework.Assert;
```
Comment (rock3r) [inline libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java:44]: Sigh, in production code 😭

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java [hunk 7]
```diff
@@ -35,49 +36,47 @@
...
+
 public abstract class AbstractBaseDrupalEntity implements DrupalClient.OnResponseListener, ICharsetItem
 {
-	transient private DrupalClient drupalClient; 
+	transient private DrupalClient drupalClient;
 
 	transient private Snapshot snapshot;
 
 	/**
 	 * In case of request canceling - no method will be triggered.
```
Comment (rock3r) [inline libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java:53]: Can we nuke this?

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java [hunk 8]
```diff
@@ -188,35 +128,15 @@ public ResponseData pullFromServer(boolean synchronous, Object tag, OnEntityRequ
...
-	 * @param listener 
-	 * @return @class ResponseData entity, containing server response or error
-	 *         in case of synchronous request, null otherwise
-	 */
-	public ResponseData deleteFromServer(boolean synchronous, Class<?> resultClass, Object tag, OnEntityRequestListener listener)
-	{
-		Assert.assertNotNull("You have to specify drupal client in order to perform requests", this.drupalClient);
-		DrupalEntityTag drupalTag = new DrupalEntityTag(false, tag, listener);
-		ResponseData result = this.drupalClient.deleteObject(this,getRequestConfig(RequestMethod.DELETE,resultClass), drupalTag, this, synchronous);
+		;
```
Comment (rock3r) [inline libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java:131]: On the previous line

File: app/src/main/java/com/connfa/ConnfaApplication.java [hunk 9]
```diff
@@ -30,7 +28,8 @@ public void onCreate() {
 
         LAPIDBRegister.getInstance().register(this, new AppDatabaseInfo(this));
 
-        Model.createInstance(this);
+        ConnfaRepository repository = ConnfaRepository.newInstance(this);
```
Comment (rock3r) [inline app/src/main/java/com/connfa/ConnfaApplication.java:31]: `#sideEffectsBonanza`, I see

**Review by rock3r — APPROVED** [review]

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java (hunk 7 above)
Comment (fourlastor) [reply libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java:53]: Dunno, it's in the drupal sdk, I'd ignore it and just get rid of the whole folder altogether 😅

File: app/src/main/java/com/connfa/model/UpdatesManager.java [hunk 10]
```diff
@@ -58,176 +57,157 @@ public static int convertEventIdToEventModePos(int eventModePos) {
...
+    }
 
-            case FLOOR_PLANS_REQUEST_ID:
-                manager = Model.getInstance().getFloorPlansManager();
-                break;
+    private Consumer<Throwable> notifyError(final UpdateCallback callback) {
+        return new Consumer<Throwable>() {
+            @Override
+            public void accept(Throwable throwable) throws Exception {
+                Timber.e(throwable);
```
[the remark this answers is not in the document]
Comment (rock3r) [reply app/src/main/java/com/connfa/model/UpdatesManager.java:193]: _It's going down, I'm yelling timber_

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java (hunk 8 above)
Comment (fourlastor) [reply libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java:131]: Will ignore, drupal sdk :P

File: app/src/main/java/com/connfa/service/ConnfaRepository.java (hunk 5 above)
Comment (fourlastor) [reply app/src/main/java/com/connfa/service/ConnfaRepository.java]: dacb2b8

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractMultipartDrupalEntity.java [hunk 11]
```diff
@@ -38,13 +38,13 @@
...
-		super(client);		
+		super(client);
 	}
-	
+
 	@Override
 	public @NonNull
     Object getManagedData()
-	{		
+	{
```
Comment (frapontillo) [inline libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractMultipartDrupalEntity.java:47]: cmd-alt-l or I get sick

File: app/src/main/java/com/connfa/model/UpdatesManager.java (hunk 4 above)
Comment (fourlastor) [reply app/src/main/java/com/connfa/model/UpdatesManager.java:110]: aa88f06

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java [hunk 12]
```diff
@@ -450,42 +310,26 @@ public Object getPatchObject()
...
 	@NonNull
 	private Object getManagedDataChecked()
 	{
 		Assert.assertNotNull("You have to specify non null data object", this.getManagedData());
 		return getManagedData();
 	}
-	
+
 	protected final class DrupalEntityTag
 	{
```
Comment (frapontillo) [inline libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractBaseDrupalEntity.java:328]: oh jeez

**Review by frapontillo — CHANGES_REQUESTED** [review]
nope

File: libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractMultipartDrupalEntity.java (hunk 11 above)
Comment (fourlastor) [reply libraries/drupalSDK/src/main/java/com/ls/drupal/AbstractMultipartDrupalEntity.java:47]: no, drupalsdk

**Review by frapontillo — APPROVED** [review]

**Comment by tiwiz** [comment]
I think we can merve

**Pull request merged**